numpy/core/fromnumeric.py

```diff
@@ -445,6 +445,10 @@ def put(a, ind, v, mode='raise'):
     array([ 0,  1,  2,  3, -5])
 
     """
+    if not isinstance(a, np.ndarray):
```

Comment:
My preference for ducktyping would suggest using a `try`/`except`? I.e.,
```
try:
    return a.put(ind, v, mode)
except AttributeError:
    if not isinstance(a, np.ndarray):
        raise TypeError(...)
    raise
```
This should also be a little faster for the most common case, where `a` is in fact an ndarray.